### PR TITLE
mapview screen

### PR DESCRIPTION
**Pull request by at-thinhnguyen2**

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Define/String.swift [hunk 1]
```diff
@@ -54,6 +54,8 @@ extension App.String {
 
     struct Key {
         static let login = "login"
+        static let kLocationUpdateNotification = "LMLocationUpdateNotification"
```
Comment (at-linhvo) [inline MyApp/Define/String.swift]: remove

File: MyApp/Define/String.swift [hunk 2]
```diff
@@ -54,6 +54,8 @@ extension App.String {
...
+        static let kLocationInfoKey = "LMLocationInfoKey"
```
Comment (at-linhvo) [inline MyApp/Define/String.swift]: remove

File: MyApp/Model/API/Core/Session.swift [hunk 3]
```diff
@@ -12,6 +12,7 @@ final class Session {
 
     static let shared = Session()
     private let defaults = UserDefaults.standard
+    var center = NotificationCenter.default
```
Comment (at-linhvo) [inline MyApp/Model/API/Core/Session.swift]: remove

File: MyApp/View/Controller/Detail/DetailViewController.swift [hunk 4]
```diff
@@ -53,11 +55,58 @@ final class DetailViewController: ViewController {
...
+        let markLast = MKPlacemark(coordinate: pointLast.coordinate, addressDictionary: nil)
+        directionsRequest.source = MKMapItem(placemark: markFirst)
+        directionsRequest.destination = MKMapItem(placemark: markLast)
+        directionsRequest.transportType = .automobile
+        let directions = MKDirections(request: directionsRequest)
+        directions.calculate(completionHandler: { response, error in
+            guard let response = response else { return }
+            guard error == nil else { return }
+            let route = response.routes[0] as MKRoute
+            self.mapView.add(route.polyline)
```
Comment (at-linhvo) [inline MyApp/View/Controller/Detail/DetailViewController.swift]: weak self

File: MyApp/View/Controller/Detail/DetailViewController.swift [hunk 5]
```diff
@@ -53,11 +55,58 @@ final class DetailViewController: ViewController {
...
         favoriteButton.imageView?.image = #imageLiteral(resourceName: "img_favorite_while")
+        addPin()
+        drawDirect()
         cogfigFavoriteButton()
         configTableView()
-        configMapView()
+    }
+
+    // MARK - Map
+    private func addPin() {
```
Comment (at-linhvo) [inline MyApp/View/Controller/Detail/DetailViewController.swift]: func addAnnotation()

File: MyApp/View/Controller/Map/MapViewController.swift [hunk 6]
```diff
@@ -7,11 +7,106 @@
...
+    }
+
+    fileprivate func foundLocation() {
+        let locationConfig = CLLocationCoordinate2D(latitude: Config.lat, longitude: Config.long)
+        let span = MKCoordinateSpanMake(Config.span, Config.span)
+        let region = MKCoordinateRegion (center: locationConfig, span: span)
+        mapView.setRegion(region, animated: true)
+    }
+
+    private func addPin() {
```
Comment (at-linhvo) [inline MyApp/View/Controller/Map/MapViewController.swift]: rename addAnnotation

File: MyApp/View/Controller/Map/MapViewController.swift [hunk 7]
```diff
@@ -7,11 +7,106 @@
...
+        pin.animatesDrop = true
+        pin.canShowCallout = true
+        pin.leftCalloutAccessoryView = UIImageView(image: #imageLiteral(resourceName: "map_img_pin"))
+        pin.rightCalloutAccessoryView = UIButton(type: .detailDisclosure)
+        return pin
+    }
+
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView,
+                 calloutAccessoryControlTapped control: UIControl) {
+        let vc = DetailViewController()
```
Comment (at-linhvo) [inline MyApp/View/Controller/Map/MapViewController.swift]: let detailViewController =

**Pull request merged**